setup.py

```diff
@@ -62,15 +62,42 @@ def check_cmake():
     except OSError:
         return False
 
-# Extend the command line options available to the install phase.
-# These arguments must come after `install` on the command line, e.g.:
-#    python3 setup.py install --mpi --arch=skylake
-#    pip3 install --install-option '--mpi' --install-option '--arch=skylake' .
-class install_command(install):
-    user_options = install.user_options + user_options_
+
+class _command_template:
+    """
+    Override a setuptools-like command to augment the command line options.
+    Needs to appear before the command class in the class's argument list for
+    correct MRO.
+    
+    Examples
+    --------
+
+    .. code-block: python
+
+      class install_command(_command_template, install):
+          pass
+
+
+      class complex_command(_command_template, mixin1, install):
+          def initialize_options(self):
+              # Both here and in `mixin1`, a `super` call is required
+              super().initialize_options()
+              # ...
+
+    Usage
+    -----
+
+        python3 setup.py install --mpi --arch=skylake
```

Comment:
Might be a better idea to refer to `doc/install/python.rst`, it's more complete.

Reply:
There was a dev comment there before, which I replaced by a docstring on the private member `_command_template`; It won't be picked up by `sphinx` by default, but if one ever makes a dev-doc with `:private-members:` this would be included; if in that context you still want to refer, could you suggest an edit?

Reply:
I meant the `Usage` section in particular, which is not complete here, and does not really need to be, as long as that documentation lives in `doc/install/python.rst`. Autodoc won't be used for some time to come, due to the incompatibilities with Pybind.

Reply:
Ok, so I remove the `Usage` section and the old dev comment about the `install_command` is not included anymore? It was a bit out of place anyway imho.

Reply:
Yes, let's remove Usage (or refer to the docs). However, the rest of the docstring can be left in place, the example is worth keeping I would say.